Feat(toggleAll): add ability to toggle all files

Jeanmsilva89 opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please follow the template so that the reviewers can easily understand what the code changes affect -->

# Summary

<!--
Explain the **motivation** for making this change: here are some points to help you:

* What issues does the pull request solve? Please tag them so that they will get automatically closed once the PR is merged
* What is the feature? (if applicable)
* How did you implement the solution?
* What areas of the website does it impact?
-->

Feature #65

Work in Progress

Add the ability to expand and collapse all files by holding the shift key and clicking on expand/collapse on any file. 
Since the Diff Section is split into 2 sections, this function works within the scope of the section you're interacting with. So if you use this function on the done section, only files in the done section will be affected and the same is true to files not in the done section.


## Test Plan

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

## What are the steps to reproduce?

## Checklist

<!-- Check completed item, when applicable, via: [X] -->

- [x] I tested this thoroughly
- [ ] I added the documentation in `README.md` (if needed)

lucasbento commented on src/components/common/Diff/DiffSection.js:
```diff
@@ -17,36 +17,54 @@ const DiffSection = ({
-      // If it's the "done" section, it shouldn't show if it's not completed
-      if (isDoneSection !== isDiffCompleted) {
-        return null
+  const toggleAll = collapse => {
```
What do you think about having one entry on `state` (something like `areAllDiffsCollapsed`) which, when set to `true`, collapsed all diffs automatically.

We can pass it to `Diff` and do `isDiffCollapsed={areAllDiffsCollapsed || isDiffCollapsed}`, that way we avoid calling all of them and have the state of this in one single place.

Jeanmsilva89 replied:
@lucasbento It's my first time using hooks so I appreciate the feedback and also I'm taking my time to try different approaches. 

I initially didn't do what you suggested because it's a bit more complicated. Simply put if I add `isDiffCollapsed={areAllDiffsCollapsed || isDiffCollapsed}` it doesn't work because any time the `areAllDiffsCollapsed` is set **true**, it would take precedence over the local `isDiffCollapsed` and keep the Diff collapsed.

In order to get around this, I set up the `areAllDiffsCollapsed` I use `undefined`. This way I can set it to `undefined` any time I want to turn off the toggleAll functionality. For example, when a user clicks on a single Diff and breaks away from the `toggleAll` functionality I set it to undefined.

Anyways, it's a bit of workaround to get this work and I think it's after all cleaner than using ref. 

let me know what you think!

lucasbento commented:
This is really cool, thank you for doing this 🙌!

Just had a comment regarding the approach, please let me know what you think 🙂

Jeanmsilva89 commented:
@lucasbento I updated the PR to follow more-less what you suggested. And I also updated the trigger to `alt` key because that's what Github does. 

let me know your thoughts.

lucasbento commented:
@Jeanmsilva89: thank you, I'm going to review it between today and tomorrow!

lucasbento commented:
@Jeanmsilva89: is this ready to merge? just asking because of the `WIP` in the title.

Jeanmsilva89 commented:
@lucasbento I updated the tile. You can merge it.

lucasbento commented:
Thank you so much for the contribution, @Jeanmsilva89!
